Sources/Algorithms/Product.swift

```diff
@@ -119,17 +119,27 @@ extension Product2: Collection where Base1: Collection {
     Index(i1: base1.endIndex, i2: base2.startIndex)
   }
   
+  @inlinable
+  public subscript(position: Index) -> (Base1.Element, Base2.Element) {
+    (base1[position.i1], base2[position.i2])
+  }
+  
+  /// Forms an index from a pair of base indices, converting
+  /// `(i, base2.endIndex)` to `(base1.index(after: i), base2.startIndex)` if
+  /// necessary.
+  @usableFromInline
+  internal func convertIndex(_ i1: Base1.Index, _ i2: Base2.Index) -> Index {
```

Comment:
is this a "conversion" or is it more of a "normalization" or "canonicalization"?